src/LuaTransformer.ts

```diff
@@ -1727,115 +1727,77 @@ export class LuaTransformer {
+                        valueExpression = tstl.createTableIndexExpression(enumReference, otherMemberName);
+                    }
                 }
-            } else {
-                const enumTable = this.transformIdentifierExpression(enumDeclaration.name);
-                const property = tstl.createTableIndexExpression(enumTable, memberName);
-                result.push(tstl.createAssignmentStatement(property, enumMember.value, enumMember.original));
 
-                const valueIndex = tstl.createTableIndexExpression(enumTable, enumMember.value);
-                result.push(tstl.createAssignmentStatement(valueIndex, memberName, enumMember.original));
-            }
-        }
-
-        return result;
-    }
-
-    protected computeEnumMembers(
-        node: ts.EnumDeclaration
-    ): Array<{ name: ts.PropertyName; value: tstl.Expression; original: ts.Node }> {
-        let numericValue = 0;
-        let hasStringInitializers = false;
-
-        const valueMap = new Map<ts.PropertyName, ExpressionVisitResult>();
-
-        return node.members.map(member => {
-            let valueExpression: ExpressionVisitResult;
-            if (member.initializer) {
-                if (ts.isNumericLiteral(member.initializer)) {
-                    numericValue = Number(member.initializer.text);
-                    valueExpression = this.transformNumericLiteral(member.initializer);
-                    numericValue++;
-                } else if (ts.isStringLiteral(member.initializer)) {
-                    hasStringInitializers = true;
-                    valueExpression = this.transformStringLiteral(member.initializer);
-                } else {
-                    if (ts.isIdentifier(member.initializer)) {
-                        const [isEnumMember, originalName] = tsHelper.isEnumMember(node, member.initializer);
-                        if (isEnumMember === true && originalName !== undefined) {
-                            if (valueMap.has(originalName)) {
-                                valueExpression = valueMap.get(originalName)!;
-                            } else {
-                                throw new Error(`Expected valueMap to contain ${originalName}`);
-                            }
-                        } else {
-                            valueExpression = this.transformExpression(member.initializer);
-                        }
-                    } else {
-                        valueExpression = this.transformExpression(member.initializer);
-                    }
+                if (!valueExpression) {
+                    valueExpression = this.transformExpression(member.initializer);
                 }
-            } else if (hasStringInitializers) {
-                throw TSTLErrors.HeterogeneousEnum(node);
             } else {
-                valueExpression = tstl.createNumericLiteral(numericValue);
-                numericValue++;
+                valueExpression = tstl.createNilLiteral();
```

Comment:
Looking at this again, is this actually correct? In what case would an enum value be nil?

Reply:
Original transformer has the same branch: https://github.com/microsoft/TypeScript/blob/master/src/compiler/transformers/ts.ts#L2442.
I have thought it'd be used for some syntactically incorrect enums, but couldn't find any cases where TS goes there.